apps/api/src/utils/aliases.ts

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2025 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import {
+	type ApplicationContentUpdates,
+	type ApplicationSignatureUpdate,
+	type CollaboratorRecord,
+	type JoinedApplicationRecord,
+} from '@/service/types.js';
+import {
+	type ApplicationContentsResponse,
+	type ApplicationResponseData,
+	type GetCollaboratorsResponse,
+	type SignatureDTO,
+} from '@pcgl-daco/data-model/src/types.js';
+import { type UpdateEditApplicationRequest } from '@pcgl-daco/validation';
+import lodash from 'lodash';
+
+/** Used in filter functions for alias utilities below
+ * @param key Current key to validate
+ * @param omittedKeys List of keys to remove from output
+ */
+const filterOmittedKeys = (key: string, omittedKeys: string[]) => !omittedKeys.includes(key);
+
+/**
+ * Helper function to convert Postgres snake_case to FE camelCase
+ * Generics allow usage w/ multiple input/output combinations
+ * @param data Original Database record with snake case keys to convert
+ * @param omittedKeys List of keys to remove from output for cases where a partial record is returned
+ * @returns ResponseRecord - Generic type representing new record with updated camelCase keys
+ */
+export const aliasToResponseData = <
+	DatabaseRecord extends Record<string, any>,
+	ResponseRecord extends Record<string, any>,
+>(
+	data: DatabaseRecord,
+	omittedKeys: string[] = [],
+): ResponseRecord => {
+	type snakeCaseKey = string & keyof DatabaseRecord;
```

Comment:
can move the type outside of the function

Reply:
OK made multiple updates related to your feedback
- imports
- move type definition
- remove map and use of String
- refactor filterOmittedKeys

Reply:
I am going to leave this unresolved b/c the imports are indeed still an issue

Reply:
Something about this Lodash import bit doesn't work as advertised

Using `import lodash from 'lodash/snakeCase';` yields `Cannot find module 'lodash/snakeCase'`
Using `require` as suggested in the Docs yields an ESM error pushing you to use `import` instead

This is an API function so I don't think we need to be super concerned about dependency size here
So I've reverted to what works `import lodash;` `lodash.camelCase()` and sticking with that

Reply:
and have renamed the two core utility functions to `convertToCamelCase` and `convertToSnakeCase` so it's extremely clear what they do